From 5e69d3cf7817af911a8a0adda276727f6ab275c5 Mon Sep 17 00:00:00 2001
From: Nadav Hugim <nhugim@habana.ai>
Date: Tue, 25 Jan 2022 13:30:28 +0200
Subject: [PATCH] [SW-62217] fit: fix bug with mismatch enum

fix mismatch of enum fit_load_result and enum fit_format_result
add to error var: fit_load_error, fit_format_error

loading invalid image will print message to uart
"FIT loading failed"

Change-Id: I6ef8f5daee4668885f7e4cbeeb07f91cdbf8f02c
Signed-off-by: Nadav Hugim <nhugim@habana.ai>
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 27d0854..82b37de 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -950,19 +950,25 @@
 int fit_load_images(void *fit)
 {
 	enum fit_sec_failure sec_res = FIT_SEC_NONE;
-	int err;
+	enum fit_load_result fit_load_error;
+	enum fit_format_result fit_format_error;
 
 	preboot_secondary_cpu_stop();
 
 	memset(&images, 0x0, sizeof(images));
 
-	err = fit_check_format(fit);
-	if (!err)
-		err = fit_copy_images(fit, &sec_res);
+	fit_format_error = fit_check_format(fit);
+	if (fit_format_error) {
+		fit_format_print_error(fit_format_error);
+		fit_load_error = FIT_LOAD_CORRUPTED;
+	} else {
+		fit_load_error = fit_copy_images(fit, &sec_res);
+	}
+
 
 	cpu_boot_err_clrset(BIT_ULL(CPU_BOOT_ERR_FIT_CORRUPTED) |
 			    BIT_ULL(CPU_BOOT_ERR_SECURITY_FAIL), false);
-	switch (err) {
+	switch (fit_load_error) {
 	case FIT_LOAD_SUCCESS:
 		return FIT_LOAD_SUCCESS;
 	case FIT_LOAD_SEC_FAILED:
@@ -972,14 +978,14 @@
 		break;
 	default:
 		cpu_boot_err_set(CPU_BOOT_ERR_FIT_CORRUPTED);
-		preboot_error_inform(HL_FW_ERR_FIT, err);
+		preboot_error_inform(HL_FW_ERR_FIT, fit_load_error);
 		break;
 	}
 
-	fit_report_err(err, sec_res);
-	fit_load_print_error(err);
+	fit_report_err(fit_load_error, sec_res);
+	fit_load_print_error(fit_load_error);
 
-	return err;
+	return fit_load_error;
 }
 
 __weak void preboot_finalize_fw_info(void) {}
